src/pybind/mgr/dashboard/controllers/cephfs.py

```diff
@@ -382,6 +382,12 @@ def ls_dir(self, fs_id, path=None, depth=1):
             paths = cfs.ls_dir(path, depth)
         except (cephfs.PermissionError, cephfs.ObjectNotFound):  # pragma: no cover
             paths = []
+        except cephfs.OSError as err:
+            location = f' at {err.args[2]}' if len(err.args) > 2 else ''
```

Comment:
If we are not re rasing the OSError in  opendir() services/cephfs.py then this might just work:
``` py
#in src/pybind/mgr/dashboard/services/cephfs.py -> opendir()
except cephfs.OSError as e:
    raise DashboardException(http_status_code=400, msg=f'{err.strerror} at {dirpath}' component='cephfs') from e
``` 
And then raising this  DashBoradException again in controller/cephfs.py to get a toasty like this:

![toasty](https://user-images.githubusercontent.com/48375025/152971873-0bd5576d-2f27-4b36-9da5-af6e6ffaeb6a.png)

What do you think?

Reply:
I don't think it's bad. It fulfills all (or almost) needed for an error message: what's failing (CephFS opendir /testfs), why (permission denied). The only (bonus) thing missing would be the "how to fix it", but the cost-benefit of adding that here does not make it worth.

So I'm fine with that message. Alternatively, you can create a higher-level (more fine-grained) exception that allows specifying the `filename`, as the Python `OSError` one.

BTW, do you need to capture and reraise the exception twice, is it not enough with letting that surface from the inner service?

Reply:
In general in OOP languages it's always desirable to encapsulate/hide logic inside classes, rather than spreading it. Thus my suggestion to use a factory class method `DashboardCephfsError.from_ceph_error(e)`.